storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -96,57 +142,119 @@ func (s *commitLogSource) Read(
 	defer iter.Close()
 
 	var (
-		namespace   = ns.ID()
-		unmerged    = make(map[uint32]map[ts.Hash]encoderMap)
-		bopts       = s.opts.ResultOptions()
-		blopts      = bopts.DatabaseBlockOptions()
-		blockSize   = ns.Options().RetentionOptions().BlockSize()
-		encoderPool = bopts.DatabaseBlockOptions().EncoderPool()
-		errs        = 0
+		namespace    = ns.ID()
+		highestShard = s.findHighestShard(shardsTimeRanges)
```

Comment:
maybe just make this `numShards = s.findHighestShard(...) + 1` and use numShards in the two places below?

Reply:
good idea, fixed